libraries/AP_Motors/AP_MotorsPulsing_Coax.cpp

```diff
@@ -0,0 +1,198 @@
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Math/AP_Math.h>
+#include "AP_MotorsPulsing_Coax.h"
+#include <GCS_MAVLink/GCS.h>
+#include <SRV_Channel/SRV_Channel.h>
+
+extern const AP_HAL::HAL& hal;
+
+const AP_Param::GroupInfo AP_MotorsPulsing_Coax::var_info[] = {
+    AP_NESTEDGROUPINFO(AP_MotorsMulticopter, 0),
+    // @Param: YAW_DIR
+    // @DisplayName: Motor normal or reverse
+    // @Description: Used to change motor rotation directions without changing wires
+    // @Values: 1:normal,-1:reverse
+    // @User: Standard
+    AP_GROUPINFO("YAW_DIR", 1, AP_MotorsPulsing_Coax, _yaw_dir, 1),
+
+    AP_GROUPEND
+};
+// init
+void AP_MotorsPulsing_Coax::init(motor_frame_class frame_class, motor_frame_type frame_type)
+{
+    // 1 - Bottom Throttle
+    // 2 - Top Throttle
+    // 3 - Bottom Pitch
+    // 4 - Top Pitch
+    // 5 - Bottom Roll
+    // 6 - Top Roll 
+    // make sure 4 output channels are mapped
+    for (uint8_t i = 0; i < 6; i++) {
+        add_motor_num(CH_1 + i);
+    }
+
+
+    // setup actuator scaling
+    motor_enabled[AP_MOTORS_MOT_1] = true;
+    motor_enabled[AP_MOTORS_MOT_2] = true;
+    motor_enabled[AP_MOTORS_MOT_3] = false;
+    motor_enabled[AP_MOTORS_MOT_4] = false;
+    motor_enabled[AP_MOTORS_MOT_5] = false;
+    motor_enabled[AP_MOTORS_MOT_6] = false;
+
+    SRV_Channels::set_angle(SRV_Channels::get_motor_function(AP_MOTORS_MOT_3), AP_MOTORS_COAX_SERVO_INPUT_RANGE);
+    SRV_Channels::set_angle(SRV_Channels::get_motor_function(AP_MOTORS_MOT_4), AP_MOTORS_COAX_SERVO_INPUT_RANGE);
+    SRV_Channels::set_angle(SRV_Channels::get_motor_function(AP_MOTORS_MOT_5), AP_MOTORS_COAX_SERVO_INPUT_RANGE);
+    SRV_Channels::set_angle(SRV_Channels::get_motor_function(AP_MOTORS_MOT_6), AP_MOTORS_COAX_SERVO_INPUT_RANGE);
+
+
+    _mav_type = MAV_TYPE_QUADROTOR;
+    rpm = AP_RPM::get_singleton();
```

Comment:
this is unused?

Reply:
Hold over from the single rotor one which was done first. You're right and it can be removed